crates/dojo-world/src/utils.rs

```diff
@@ -328,6 +332,55 @@ pub fn block_number_from_receipt(receipt: &TransactionReceipt) -> u64 {
     }
 }
 
+#[allow(async_fn_in_trait)]
+pub trait TransactionExt<T>
+where
+    T: ConnectedAccount + Sync,
+{
+    type R;
+
+    async fn send_with_cfg(
+        self,
+        txn_config: &TxnConfig,
+    ) -> Result<Self::R, AccountError<T::SignError>>;
+}
+
+impl<T> TransactionExt<T> for Execution<'_, T>
```

Comment:
```suggestion
impl<T> ExecutionExt<T> for Execution<'_, T>
```
Following up the discussion, `ExecutionExt` may be more appropriate? As a transaction in the stack usually refer to the type containing the data itself.

@kariy @lambda-0x thoughts on that?

Reply:
naming it `ExecutionExt` would give an impression that its only relevant for `invoke` transactions but it can be implemented for `declare` transactions and maybe even for `deploy_account` transactions in future.

so basically this extension trait can be used for [all types of transactions in starknet](https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/transactions/), so i named it `TransactionExt`